From d5363d4ed2c4604d4e1604b4531a30540b32f32d Mon Sep 17 00:00:00 2001
From: Nicolas Toromanoff <nicolas.toromanoff@foss.st.com>
Date: Thu, 31 Mar 2022 09:31:23 +0200
Subject: [PATCH] ARM: dts: stm32: add dmas entries for CRYP driver

The CRYP IP can use the MDMA to transfer in and out buffers.
The DMA can be used in CRYP for stm32mp13xc, stm32mp13xf, stm32mp15xc,
stm32mp15xf and can be enabled by default to gain performance.

Signed-off-by: Nicolas Toromanoff <nicolas.toromanoff@foss.st.com>
Change-Id: Ic552beb9046ac3ff855a65bc960be5f301bd4c69
---
 arch/arm/boot/dts/st/stm32mp13xc.dtsi | 3 +++
 arch/arm/boot/dts/st/stm32mp13xf.dtsi | 3 +++
 arch/arm/boot/dts/st/stm32mp15xc.dtsi | 3 +++
 arch/arm/boot/dts/st/stm32mp15xf.dtsi | 3 +++
 4 files changed, 12 insertions(+)

diff --git a/arch/arm/boot/dts/st/stm32mp13xc.dtsi b/arch/arm/boot/dts/st/stm32mp13xc.dtsi
index 4d00e7592882..fc4ba53fecaa 100644
--- a/arch/arm/boot/dts/st/stm32mp13xc.dtsi
+++ b/arch/arm/boot/dts/st/stm32mp13xc.dtsi
@@ -12,6 +12,9 @@ cryp: crypto@54002000 {
 			interrupts = <GIC_SPI 80 IRQ_TYPE_LEVEL_HIGH>;
 			clocks = <&rcc CRYP1>;
 			resets = <&rcc CRYP1_R>;
+			dmas = <&mdma 28 0x0 0x400202 0x0 0x0 0x0>,
+			       <&mdma 29 0x3 0x400808 0x0 0x0 0x0>;
+			dma-names = "in", "out";
 			status = "disabled";
 		};
 	};
diff --git a/arch/arm/boot/dts/st/stm32mp13xf.dtsi b/arch/arm/boot/dts/st/stm32mp13xf.dtsi
index 4d00e7592882..fc4ba53fecaa 100644
--- a/arch/arm/boot/dts/st/stm32mp13xf.dtsi
+++ b/arch/arm/boot/dts/st/stm32mp13xf.dtsi
@@ -12,6 +12,9 @@ cryp: crypto@54002000 {
 			interrupts = <GIC_SPI 80 IRQ_TYPE_LEVEL_HIGH>;
 			clocks = <&rcc CRYP1>;
 			resets = <&rcc CRYP1_R>;
+			dmas = <&mdma 28 0x0 0x400202 0x0 0x0 0x0>,
+			       <&mdma 29 0x3 0x400808 0x0 0x0 0x0>;
+			dma-names = "in", "out";
 			status = "disabled";
 		};
 	};
diff --git a/arch/arm/boot/dts/st/stm32mp15xc.dtsi b/arch/arm/boot/dts/st/stm32mp15xc.dtsi
index f729b0d1bd96..6519b0731c01 100644
--- a/arch/arm/boot/dts/st/stm32mp15xc.dtsi
+++ b/arch/arm/boot/dts/st/stm32mp15xc.dtsi
@@ -14,6 +14,9 @@ cryp1: cryp@54001000 {
 			interrupts = <GIC_SPI 79 IRQ_TYPE_LEVEL_HIGH>;
 			clocks = <&rcc CRYP1>;
 			resets = <&rcc CRYP1_R>;
+			dmas = <&mdma1 29 0x0 0x400202 0x0 0x0 0x0>,
+			       <&mdma1 30 0x3 0x400808 0x0 0x0 0x0>;
+			dma-names = "in", "out";
 			status = "disabled";
 		};
 	};
diff --git a/arch/arm/boot/dts/st/stm32mp15xf.dtsi b/arch/arm/boot/dts/st/stm32mp15xf.dtsi
index 82288c3d8676..3090693e1332 100644
--- a/arch/arm/boot/dts/st/stm32mp15xf.dtsi
+++ b/arch/arm/boot/dts/st/stm32mp15xf.dtsi
@@ -14,6 +14,9 @@ cryp1: cryp@54001000 {
 			interrupts = <GIC_SPI 79 IRQ_TYPE_LEVEL_HIGH>;
 			clocks = <&rcc CRYP1>;
 			resets = <&rcc CRYP1_R>;
+			dmas = <&mdma1 29 0x0 0x400202 0x0 0x0 0x0>,
+			       <&mdma1 30 0x3 0x400808 0x0 0x0 0x0>;
+			dma-names = "in", "out";
 			status = "disabled";
 		};
 	};
-- 
2.39.5

